### PR TITLE
Adds support for OpenSUSE/SLES/SLED, improves distribution detection.

### PR DESCRIPTION
**Pull request by colstrom**

### CODE REVIEW

File: src/linux/install-tools.sh [hunk 1]
```diff
@@ -54,7 +54,10 @@ install_tools() {
   cp -r ./usr/sbin/mdata-* /usr/sbin/
   cp -r ./usr/share/man/man1/mdata-* /usr/share/man/man1/
   ln -fs /usr/sbin/mdata-get /lib/smartdc/mdata-get
-  mv /etc/rc.local /etc/rc.local-backup
+  if [[ -e /etc/rc.local ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: This is a nice addition! However I'd prefer this to be consistent to how we typically format test blocks, e.g.,:

```
if [[ -e /etc/rc.local ]] ; then
```

File: src/linux/install-tools.sh [hunk 2]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
-    if [[ -f /etc/redhat-release ]] ; then
-      install_redhat
-    elif [[ -f /etc/debian_version ]] ; then
-      install_debian
-    else
-      fatal "Sorry. Your OS ($OS) is not supported by this installer"
-    fi
-    ;;
-  *)
+if [[ "$(uname -s)" != "Linux" ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: Per previous comment, let's change this to 

```
if [[ "$(uname -s)" != "Linux" ]] ; then
```

File: src/linux/install-tools.sh [hunk 3]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
+then
     fatal "Sorry. Your OS ($OS) is not supported by this installer"
-    ;;
-esac
+fi
+
+if [[ -f /etc/os-release ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: See previous comment about formatting test block

File: src/linux/install-tools.sh [hunk 4]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
+    fatal "Sorry. Your OS ($ID) is not supported by this installer"
+  fi
+else
+  if [[ -f /etc/redhat-release ]]
+  then
+    install_redhat
+  elif [[ -f /etc/debian_version ]]
+  then
+    install_debian
+  elif [[ -f /etc/SuSE-release ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: I'm not too familiar with SUSE, but is this file really in mixed caps?

File: src/linux/install-tools.sh [hunk 5]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
+  method=unknown
+
+  for distribution in ${ID_LIKE} ${ID}
+  do
+    case ${distribution} in
+      debian|ubuntu)           method=debian;;
+      fedora|centos|rhel)      method=redhat;;
+      suse|opensuse|sles|sled) method=suse;;
+    esac
+  done
```
Comment (chorrell) [inline src/linux/install-tools.sh]: This formatting would be preferable:

for distribution in ${ID_LIKE} ${ID}; do
    case ${distribution} in
      debian|ubuntu)
        method=debian
        ;;
      fedora|centos|rhel)
        method=redhat
        ;;
      suse|opensuse|sles|sled)
        method=suse
        ;;
    esac
done

File: src/linux/install-tools.sh [hunk 6]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
+
+  if [[ -n "${method}" ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: See previous comment about test/if blocks

File: src/linux/install-tools.sh [hunk 7]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
+  then
+    eval install_${method}
+  else
+    fatal "Sorry. Your OS ($ID) is not supported by this installer"
+  fi
+else
+  if [[ -f /etc/redhat-release ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: See previous comment about test/if blocks

File: src/linux/install-tools.sh [hunk 8]
```diff
@@ -91,22 +106,46 @@ if [[ $1 != "-y" ]]; then
...
+  then
+    install_redhat
+  elif [[ -f /etc/debian_version ]]
```
Comment (chorrell) [inline src/linux/install-tools.sh]: See previous comment about test/if blocks

File: src/linux/install-tools.sh (hunk 4 above)
Comment (chorrell) [inline src/linux/install-tools.sh]: See previous comment about test/if blocks

**Review by chorrell — CHANGES_REQUESTED** [review]
Some formatting nits and one question about mixed caps in /etc/SuSE-release

File: src/linux/install-tools.sh (hunk 1 above)
Comment (colstrom) [reply src/linux/install-tools.sh]: Gotcha. I'll amend my patch accordingly.

File: src/linux/install-tools.sh (hunk 5 above)
Comment (colstrom) [reply src/linux/install-tools.sh]: Mmm, was that meant to format differently? I'd expect indentation within the `case`, no?

File: src/linux/install-tools.sh (hunk 4 above)
Comment (colstrom) [reply src/linux/install-tools.sh]: I did a double-take when I saw it too, but it is correct (verified on SLES 12):
```
 $ ls -l /etc/S*
-rw-r--r--   1 root     root           17 Oct 25  2016 /etc/SUSE-brand
-rw-r--r--   1 root     root           46 Apr  1 15:05 /etc/SUSEConnect
-rw-r--r--   1 root     root          377 Mar  6 02:06 /etc/SUSEConnect.example
-rw-r--r--   1 root     root          215 Oct 25  2016 /etc/SuSE-release
```

File: src/linux/install-tools.sh (hunk 5 above)
Comment (chorrell) [reply src/linux/install-tools.sh]: Yes, correct, this was a markdown formatting issue on my part.

**Review by chorrell — APPROVED** [review]

**Comment by chorrell** [comment]
LGTM

**Pull request merged**